lib/DidWebDriver.js

```diff
@@ -0,0 +1,161 @@
+        doc: didDocument, purpose
+      });
+      return keyPairs.get(methodId);
+    };
+
+    return {didDocument, keyPairs, methodFor};
+  }
+
+  /**
+   * Returns the public key (verification method) object for a given DID
+   * Document and purpose. Useful in conjunction with a `.get()` call.
+   *
+   * @example
+   * const didDocument = await didKeyDriver.get({did});
+   * const authKeyData = didDriver.publicMethodFor({
+   *   didDocument, purpose: 'authentication'
+   * });
+   * // You can then create a suite instance object to verify signatures etc.
+   * const authPublicKey = await cryptoLd.from(authKeyData);
+   * const {verify} = authPublicKey.verifier();
+   *
+   * @param {object} options - Options object.
+   * @param {object} options.didDocument - DID Document (retrieved via a
+   *   `.get()` or from some other source).
+   * @param {string} options.purpose - Verification method purpose, such as
+   *   'authentication', 'assertionMethod', 'keyAgreement' and so on.
+   *
+   * @returns {object} Returns the public key object (obtained from the DID
+   *   Document), without a `@context`.
+   */
+  publicMethodFor({didDocument, purpose} = {}) {
+    if(!didDocument) {
+      throw new TypeError('The "didDocument" parameter is required.');
+    }
+    if(!purpose) {
+      throw new TypeError('The "purpose" parameter is required.');
+    }
+
+    const method = didIo.findVerificationMethod({doc: didDocument, purpose});
+    if(!method) {
+      throw new Error(`No verification method found for purpose "${purpose}"`);
+    }
+    return method;
+  }
+
+  /**
+   * Returns a `did:web` method DID Document for a given DID, or a key document
+   * for a given DID URL (key id).
+   * Either a `did` or `url` param is required.
+   *
+   * @example
+   * await resolver.get({did}); // -> did document
+   * await resolver.get({url: keyId}); // -> public key node
+   *
+   * @param {object} options - Options object.
+   * @param {string} [options.did] - DID or a key id (either an ed25519 key
+   *   or an x25519 key-agreement key id).
+   * @param {string} [options.url] - Alias for the `did` url param, supported
+   *   for better readability of invoking code.
+   * @param {object} [options.fetchOptions = {}] - Options for the request.
+   *
+   * @returns {Promise<object>} Resolves to a DID Document or a
+   *   public key node with context.
+   */
+  async get({did, url, fetchOptions = {}} = {}) {
+    did = did || url;
+    if(!did) {
+      throw new TypeError('"did" must be a non-zero length string.');
+    }
+    // overwrite global options with request specific options
+    const requestOptions = {...this.fetchOptions, ...fetchOptions};
+    const {fullUrl, fragment} = didToUrl(did);
+    assertDomain({allowList: this.allowList, url: fullUrl});
+    const {data} = await httpClient.get(fullUrl, requestOptions);
+    const [publicKeyDescription] = data.verificationMethod;
+    const [didAuth] = did.split('#');
```

Comment:
If the did contains a query like `?service=foo` this will fail, so need to account for fragment and query here.

Reply:
Addressed here: https://github.com/digitalbazaar/did-method-web/pull/1/commits/6ff50f18094bc8abe49e6096f9e266e4e200070b